docs/DeveloperGuide.md

```diff
@@ -137,55 +137,131 @@ Classes used by multiple components are in the `seedu.clinic.commons` package.
 
-Step 4. If the user wants to delete an entire warehouse/supplier entry, `model#deleteWarehouse` will remove the entry
-from the list in the `model`.
+#### What Delete Feature does
 
-If the user wants to delete a product inside the entry, the set of `Product` for the warehouse/supplier entry will be retrieved first.
-The `warehouse#getProductByName` or `supplier#getProductByName` will give the target product to delete from the name argument,
-and the retrieved product will be removed from the current product set in the selected warehouse/supplier entry.
-Afterwards, the updated product set will replace the existing set in this warehouse/supplier. The selected warehouse/supplier entry in the model is then 
-replaced by the updated warehouse/supplier with the target product deleted.
+The `delete` feature allows user to remove a warehouse or supplier __(case 1)__.
+<br>
+The `delete` feature also allows user to remove a product from a certain warehouse or supplier __(case 2)__.
+<br>
+<br>
+In case 1, `TYPE` needs to be set to `w` or `s`.
+<br>
+In case 2, `TYPE` needs to be set to `pw` or `ps`, and a `PRODUCT_NAME` needs to be specified.
+<br>
+<br>
+The deletion is limited to the items shown in the list displayed in GUI, and is done one item at a time.
+
+#### Path Execution of Delete Command
+
+The workflow of an `Delete` Command when it is executed by a user is shown in the activity diagram below:
+
+![Delete Command Activity Diagram](images/DeleteCommandActivityDiagram.png)
+<center><i>Figure n. Delete Command Activity Diagram</i></center>
+<br>
+
+With reference to the activity diagram above, the user input will be sent to **`DeleteCommandParser`** for parsing.
+The **`DeleteCommandParser`** will check if the compulsory prefixes are present (i.e. `ct/TYPE` and `i/INDEX`) and the corresponding
+argument values are all valid.
+A **`ParseException`** will be thrown if the check fails.
+
+If the `TYPE` parsed indicates a product deletion (via `ct/ps` or `ct/pw`), an additional field `PRODUCT_NAME` is parsed with its prefix and value checked.
+
+Next, a new **`DeleteCommand`** will be generated and executed. There are four possible paths for the `delete` command:
+
+1. The target to delete is a `supplier`/`warehouse` <br>
+CLI-nic finds the target `supplier`/`warehouse` at the specified `INDEX` of the displayed list, and remove it completely.
+
+1. The `INDEX` specified is invalid (e.g. exceeds the length of the list) <br>
+A **`CommandException`** error message wil be thrown.
+
+1. The target to delete is a `product` in a particular 'supplier' <br>
+CLI-nic finds the target `supplier` at the specified `INDEX` of the displayed supplier list, and retrieve its product list.<br>
+It finds the `product` with specified `PRODUCT_NAME` and remove it from the product list.
+
+1. No `product` has the `PRODUCT_NAME` in the target warehouse/supplier <br>
+A **`CommandException`** error message wil be thrown.
```

Comment:
Should the numbering be changed to "1. 2. 3. 4." ?

Reply:
Nope, this is the markdown standard actually.

Reply:
Oh okay, I didn't know that, sorry about that!:)